### ISSUE TITLE
Cross Cluster Service calls are failing (Success only with namespace within the definition)

### ISSUE THREAD

**Issue by vikram-yerneni** [istio/istio.io#12907]
Kubernetes Clusters: Primary/Secondary Setup
ISTIO: Primary/Remote Setup Enabled       (https://istio.io/latest/docs/setup/install/multicluster/primary-remote_multi-network/?_ga=2.47427886.827391038.1678826508-548589853.1678826508)

The cross service calls are failing without the namespace reference in a simple curl calls.
Not sure what am I missing here
"curl -sS service.namespace:5000/hello" vs "curl -sS service:5000/hello"

Communication within the cluster is working fine and in cross cluster communication is working when the namespace is provided. Without the namespace the communication is throwing an  error :
curl: (6) Could not resolve host: helloworld

Basically, We are trying to hide namespace from the endpoints to provide location-transparency to our customers.
We tried ExternalName Service to reduce
curl -sS service.namespace:5000/hello1  to curl -sS service:5000/hello , but ExternalName is not working across cluster.
